apps/storefront/src/gatsby-theme-docz/components/Sidebar/index.js

```diff
@@ -123,15 +124,18 @@ export const Sidebar = ({ className, open, onClick }) => {
   const routesWithTwoSegments = (doc) =>
     doc.route.replace(/^\/|\/$/g, '').split('/').length < 3
 
-  const subMenus = useMemo(() =>
-    docs
-      .filter(notRootRoute)
-      .filter(routesWithTwoSegments)
-      .map((doc) => ({
-        title: doc.title,
-        route: doc.route,
-        current: current.route.includes(doc.route),
-      })),
+  const handleDrafts = (doc) =>
+    !(process.env.GATSBY_STAGE === 'prod' && doc.mode === 'draft')
+
+  const subMenus = useMemo(
+    () =>
+      R.pipe(
+        R.filter(notRootRoute),
+        R.filter(routesWithTwoSegments),
+        R.filter(handleDrafts),
+        R.reduce((acc, { title, route }) => [...acc, { title, route }], []),
+      )(docs),
+    [docs],
```

Comment:
Not sure there’s any real benefit to using Ramda here instead of native filter and reduce, but one thing you _can_ do which is quite nice is to intercept a piped function with R.tap(console.log) to see what’s happening at that particular point. Also since we’re using Ramda extensively in Figma Broker, we might as well get into the habbit of using it in other places as well – it’s a powerful library.

Reply:
Since you are using Ramda, maybe the `R.reduce` for picking props can be replaced with [R.pick](https://ramdajs.com/docs/#pick) ?